### PR TITLE
Implements operator and CSI subcommand

### PR DESCRIPTION
**Pull request by meik99**
# Description

Reimplements the operator subcommands as Cobra commands.

N.B.: I moved the original files from src/cmd/operator to src/cmd/ but added new ones with the same name. That is why some old files are listed as new ones

## How can this be tested?

* Deploy the DTO on a cluster
* All DTO components must start as usual and run as before

## Checklist
- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly
- [x] I have read and understood the [contribution guidelines](/CONTRIBUTING.md)

### CODE REVIEW

File: src/cmd/operator/manager.go [hunk 1]
```diff
@@ -1,63 +1,102 @@
...
+
+	leaderElectionId           = "dynatrace-operator-lock"
+	leaderElectionResourceLock = "configmaps"
+
+	livenessEndpointName = "/livez"
+	readyzEndpointName   = "readyz"
+	livezEndpointName    = "livez"
+)
+
+type managerProvider interface {
```
Comment (0sewa0) [inline src/cmd/operator/manager.go:25]: why not use https://github.com/Dynatrace/dynatrace-operator/pull/876/files#diff-ac22401e6b044dcfe9a455ef86f78e3c149743333788c8912b83a6a4b1ef9cc6R8 instead ?

Comment (meik99) [reply src/cmd/operator/manager.go:25]: cause that's a mocked provider, the link is a mocked manager

File: src/cmd/config/mock_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/rest"
+)
+
+func TestMockConfigProvider(t *testing.T) {
```
Comment (mjgrzybek) [inline src/cmd/config/mock_test.go:11]: Why test mocks?

File: src/cmd/csi_driver.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+//		return nil, cleanUp, err
+//	}
+//
+//	if err := csigc.NewCSIGarbageCollector(mgr.GetClient(), csiOpts, access).SetupWithManager(mgr); err != nil {
+//		log.Error(err, "unable to create CSI Garbage Collector")
+//		return nil, cleanUp, err
+//	}
+//
+//	return mgr, cleanUp, nil
+//}
```
Comment (mjgrzybek) [inline src/cmd/csi_driver.go:99]: Leftovers

File: src/cmd/config/mock.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package config
+
+import (
+	"github.com/stretchr/testify/mock"
+	"k8s.io/client-go/rest"
+)
+
+type MockProvider struct {
```
Comment (mjgrzybek) [inline src/cmd/config/mock.go:8]:
```suggestion
type MockProvider mock.Mock
```

File: src/cmd/config/mock.go [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+	mock.Mock
+}
```
Comment (mjgrzybek) [inline src/cmd/config/mock.go:10]:
```suggestion
type MockProvider mock.Mock
```

**Comment by mjgrzybek** [comment]
@meik99 What's the idea behind hiding methods for different structs behind `type Provider interface`?

File: src/cmd/operator.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+//
+//func setupBootstrapper(ns string, cfg *rest.Config, cancelMgr context.CancelFunc) (manager.Manager, error) {
+//	log.Info("starting certificate bootstrapper", "namespace", ns)
+//	mgr, err := setupBootstrapMgr(ns, cfg)
+//	if err != nil {
+//		return mgr, err
+//	}
+//
+//	return mgr, certificates.AddBootstrap(mgr, ns, cancelMgr)
+//}
```
Comment (mjgrzybek) [inline src/cmd/operator.go:32]: Leftovers

File: src/cmd/standalone.go [hunk 7]
```diff
@@ -0,0 +1,9 @@
+package main
+
+//func startStandAloneInit() error {
+//	standaloneRunner, err := standalone.NewRunner(afero.NewOsFs())
+//	if err != nil {
+//		return err
+//	}
+//	return standaloneRunner.Run()
+//}
```
Comment (mjgrzybek) [inline src/cmd/standalone.go:9]: Leftovers

File: src/cmd/validation.go [hunk 8]
```diff
@@ -14,16 +14,11 @@ limitations under the License.
...
-	return mgr, cleanUp, nil
-}
+//func startValidationServer(mgr manager.Manager) (manager.Manager, func(), error) {
+//	cleanUp := func() {}
+//	if err := validationhook.AddDynakubeValidationWebhookToManager(mgr); err != nil {
+//		return nil, cleanUp, err
+//	}
+//
+//	return mgr, cleanUp, nil
+//}
```
Comment (mjgrzybek) [inline src/cmd/validation.go:24]: Leftovers

File: src/cmd/main.go [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+	"os"
+
+	cmdConfig "github.com/Dynatrace/dynatrace-operator/src/cmd/config"
+	"github.com/Dynatrace/dynatrace-operator/src/cmd/operator"
+	"github.com/Dynatrace/dynatrace-operator/src/cmd/webhook"
+	"github.com/Dynatrace/dynatrace-operator/src/kubesystem"
+	"github.com/Dynatrace/dynatrace-operator/src/logger"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
```
Comment (mjgrzybek) [inline src/cmd/main.go:27]:
```suggestion
```

**Review by mjgrzybek — CHANGES_REQUESTED** [review]

File: src/cmd/config/mock_test.go (hunk 2 above)
Comment (meik99) [reply src/cmd/config/mock_test.go:11]: I used TDD for the most parts, so the tests came first

File: src/cmd/csi_driver.go (hunk 3 above)
Comment (meik99) [reply src/cmd/csi_driver.go]: kept for reference, should've marked it as a draft. Leftovers are deleted now

**Comment by meik99** [comment]
> @meik99 What's the idea behind hiding methods for different structs behind `type Provider interface`?

Making the command builders and built commands more testable.
By using interfaces instead of specific classes they can be more easily exchanged for mocks

File: src/cmd/config/mock.go (hunk 5 above)
Comment (meik99) [reply src/cmd/config/mock.go:10]: Results in unresolved reference errors when calling the mock functions, e.g. `Called`.
It's also used like that in the [docs](https://pkg.go.dev/github.com/stretchr/testify/mock)

**Review by mjgrzybek — APPROVED** [review]

**Pull request merged**